tempesta_fw/ss_skb.c

```diff
@@ -788,18 +788,13 @@ __skb_fragment(struct sk_buff *skb_head, struct sk_buff *skb, char *pspt,
 		d_size = skb_frag_size(frag);
 		offset = pspt - (char *)skb_frag_address(frag);
 
-		if (offset >= 0 && offset <= d_size) {
+		/*
+		 * @pspt can be the end of a frag, but it can also be a start of
+		 * another fragment. Not necessary, that both frags are
+		 * neighbours.
+		 */
+		if (offset >= 0 && offset < d_size) {
 			int t_size = d_size - offset;
-			if (!t_size) {
-				/*
-				 * @pspt is at the end of the frag (zero tail
-				 * length): append if @len > 0 or move to the
-				 * next frag for deletion.
-				 */
-				if (len > 0)
-					goto append;
-				continue;
-			}
 			len = max(len, -t_size);
```

Comment:
We don't need `t_size` any more, just use `max_t(long, len, offset - d_size)` and the same in the `if (offset >= 0 && offset < d_size)` above